chirp/drivers/uvk5.py

```diff
@@ -0,0 +1,1339 @@
+                tx_tone = CTCSS_TONES[_mem.txcode]
+            else:
+                tx_tone = 0
+                tx_tmode = ""
+        elif tx_tmode == "DTCS":
+            if _mem.txcode < len(DTCS_CODES):
+                tx_tone = DTCS_CODES[_mem.txcode]
+            else:
+                tx_tone = 0
+                tx_tmode = ""
+
+        if rx_tmode == "Tone":
+            if _mem.rxcode < len(CTCSS_TONES):
+                rx_tone = CTCSS_TONES[_mem.rxcode]
+            else:
+                rx_tone = 0
+                rx_tmode = ""
+        elif rx_tmode == "DTCS":
+            if _mem.rxcode < len(DTCS_CODES):
+                rx_tone = DTCS_CODES[_mem.rxcode]
+            else:
+                rx_tone = 0
+                rx_tmode = ""
+
+        tx_pol = txtype == 0x03 and "R" or "N"
+        rx_pol = rxtype == 0x03 and "R" or "N"
+
+        chirp_common.split_tone_decode(mem, (tx_tmode, tx_tone, tx_pol),
+                                       (rx_tmode, rx_tone, rx_pol))
+
+    # Extract a high-level memory object from the low-level memory map
+    # This is called to populate a memory in the UI
+
+    def get_memory(self, number2):
+        number = number2-1  # in the radio memories start with 0
+
+        mem = chirp_common.Memory()
+
+        # cutting and pasting configs from different radios
+        # might try to set channel 0
+        if number2 == 0:
+            LOG.warning("Attempt to get channel 0")
+            return mem
+
+        _mem = self._memobj.channel[number]
+
+        tmpcomment = ""
+
+        mem.number = number2
+
+        # We'll consider any blank (i.e. 0MHz frequency) to be empty
+        if (_mem.freq == 0xffffffff) or (_mem.freq == 0):
+            mem.empty = True
+            # set some sane defaults:
+            mem.power = UVK5_POWER_LEVELS[2]
+            mem.extra = RadioSettingGroup("Extra", "extra")
+            rs = RadioSetting("bclo", "BCLO", RadioSettingValueBoolean(False))
+            mem.extra.append(rs)
+            rs = RadioSetting("frev", "FreqRev",
+                              RadioSettingValueBoolean(False))
+            mem.extra.append(rs)
+            rs = RadioSetting("pttid", "PTTID", RadioSettingValueList(
+                PTTID_LIST, PTTID_LIST[0]))
+            mem.extra.append(rs)
+            rs = RadioSetting("dtmfdecode", "DTMF decode",
+                              RadioSettingValueBoolean(False))
+            mem.extra.append(rs)
+            rs = RadioSetting("scrambler", "Scrambler", RadioSettingValueList(
+                SCRAMBLER_LIST, SCRAMBLER_LIST[0]))
+            mem.extra.append(rs)
+
+            # actually the step and duplex are overwritten by chirp based on
+            # bandplan. they are here to document sane defaults for IARU r1
+            # mem.tuning_step = 25.0
+            # mem.duplex = "off"
+
+            return mem
+
+        if number > 199:
+            mem.name = "VFO_"+str(number-199)
```

Comment:
You should be setting `mem.extd_number` with this value, not the name. If the memory can't have an alpha tag then you should remove this. You need to set these special channels in `RadioFeatures.valid_special_chans`. You also need to handle the number being requested as a string which will be the case for special channels.